### ISSUE TITLE
Remove inactive classes from NavLink

### ISSUE THREAD

**Issue by tim-phillips** [remix-run/react-router#7194]
Using the `NavLink` component, I'd like to remove classes that should not be on an active link. 

I'm using Tailwind and `text-gray-200` does not override `text-gray-800` when it is added to the element. I would like the class `text-gray-800` to not be present while the link is active. Perhaps an `inactiveClassName` prop. I'd be happy to start a PR with some guidance. 

What I've tried:
```js
<NavLink
  to="/"
  className="px-3 py-2 text-gray-800"
  activeClassName="rounded-sm text-gray-200 bg-blue-gray-dark"
>
  Dashboard
</NavLink>
```

Could be:
```js
<NavLink
  to="/"
  className="px-3 py-2"
  inactiveClassName="text-gray-800"
  activeClassName="rounded-sm text-gray-200 bg-blue-gray-dark"
>
  Dashboard
</NavLink>
```

**Comment by timdorr** [comment]
You can just disable the `!important` declarations: https://tailwindcss.com/docs/configuration/#important

Then the text colors won't be overridden by their order in the CSS file.

**Comment by tim-phillips** [comment]
@timdorr, that setting is disabled by default. I tried all values, including `'#root'` with no effect.

The reason I bring this up here is because this has been discussed at https://github.com/tailwindcss/tailwindcss/issues/1010. 

> If you are trying to dynamically toggle between p-0 and p-4 you should always swap them out, toggle both, you shouldn't rely on css source order.

**Comment by garand** [comment]
@timdorr Is this something you'd be willing to reconsider? I totally get adding extra bloat that doesn't make sense, but as I've been using tools like Tailwind CSS more recently as well it can become necessary to remove classes in "inactive" situations.

**Comment by timdorr** [comment]
You can use `@apply` to create custom component classes from your Tailwind classes that better match the behaviors here.

**Comment by garand** [comment]
@timdorr Yeah, I end up doing that usually but IMO it would be cleaner to only be applying the classes used on an element.

Appreciate the quick response!

**Comment by mjackson** [comment]
@chaance Fixing this would be a great candidate for another v5 update next week.

**Comment by chaance** [comment]
Just keeping everyone here in the loop, where I think we've landed is addressed in #7983. This solution is a bit lower level than simply adding `inactiveClassName` or `inactiveStyle` props, but it's a stronger base to handle either active or inactive styles with a single prop instead of splitting `className` or `style` into three separate props for each case.

The lovely thing with this approach is that, if you prefer an `inactiveClassName` prop, it's quite simple to create your own abstraction and use whatever API you'd prefer:

```jsx
import { NavLink as BaseNavLink } from "react-router-dom";
import cx from "classnames"; // this can be whatever tool you want, obviously!

const NavLink = React.forwardRef(({ inactiveClassName, ...props }, ref) => {
  return (
    <BaseNavLink
      ref={ref}
      {...props}
      className={isActive => !isActive ? cx(props.className, inactiveClassName) : props.className}
    />
  );
});
```

**Comment by timdorr** [comment]
Actually, a slightly more elegant way of doing it:
```js
import { NavLink as BaseNavLink } from "react-router-dom";
import cx from "classnames"; // this can be whatever tool you want, obviously!

const NavLink = React.forwardRef(({ activeClassName, inactiveClassName, className, ...props }, ref) => {
  return (
    <BaseNavLink
      ref={ref}
      {...props}
      className={isActive => cx({ activeClassName: isActive, inactiveClassName: !isActive }, className)}
    />
  );
});
```
I like it! 👍

**Comment by chaance** [comment]
Merged in #7983

**Comment by tim-phillips** [comment]
Nice solution! Thanks a ton!

**Comment by chaance** [comment]
Just a note that we made a tiny tweak to the API before releasing, so `isActive` is a named object property in the function instead of the parameter itself. Just released in v6.0.0-beta.3!
